Sources/Foundation/ListFormatter.swift

```diff
@@ -0,0 +1,99 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+@_implementationOnly import CoreFoundation
+
+/* NSListFormatter provides locale-correct formatting of a list of items using the appropriate separator and conjunction. Note that the list formatter is unaware of the context where the joined string will be used, e.g., in the beginning of the sentence or used as a standalone string in the UI, so it will not provide any sort of capitalization customization on the given items, but merely join them as-is. The string joined this way may not be grammatically correct when placed in a sentence, and it should only be used in a standalone manner.
+*/
+open class ListFormatter: Formatter {
+    private let cfFormatter: CFListFormatter
+
+    /* Specifies the locale to format the items. Defaults to autoupdatingCurrentLocale. Also resets to autoupdatingCurrentLocale on assignment of nil.
+     */
+    private var _locale: Locale = .autoupdatingCurrent
+    open var locale: Locale! {
+        get { return _locale }
+        set { _locale = newValue ?? .autoupdatingCurrent }
+    }
+
+    /* Specifies how each object should be formatted. If not set, the object is formatted using its instance method in the following order: -descriptionWithLocale:, -localizedDescription, and -description.
+     */
+    /*@NSCopying*/ open var itemFormatter: Formatter?
+
+    public override init() {
+        self.cfFormatter = _CFListFormatterCreate(kCFAllocatorSystemDefault, CFLocaleCopyCurrent())!
+        super.init()
+    }
+
+    public required init?(coder: NSCoder) {
+        self.cfFormatter = _CFListFormatterCreate(kCFAllocatorSystemDefault, CFLocaleCopyCurrent())!
```

Comment:
Most of the SCF does not support `initWithCoder`.  Shall we follow precedent and `fatalError()` here?